src/conf_mode/system_watchdog.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+#
+# Copyright VyOS maintainers and contributors <maintainers@vyos.io>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from sys import exit
+from pathlib import Path
+
+from vyos.config import Config
+from vyos.template import render
+from vyos.utils.process import call, run
+from vyos import ConfigError
+from vyos import airbag
+
+airbag.enable()
+
+watchdog_config_dir = Path('/run/systemd/system.conf.d')
+watchdog_config_file = Path(watchdog_config_dir / 'watchdog.conf')
+modules_load_directory = Path('/run/modules-load.d')
+modules_load_file = Path(modules_load_directory / 'watchdog.conf')
+WATCHDOG_DEV = Path('/dev/watchdog0')
+
+
+def get_config(config=None):
+    if config:
+        conf = config
+    else:
+        conf = Config()
+    base = ['system', 'watchdog']
+
+    if not conf.exists(base):
+        return None
+
+    watchdog = conf.get_config_dict(
+        base, key_mangling=('-', '_'), get_first_key=True, with_recursive_defaults=True
+    )
+
+    return watchdog
+
+
+def verify(watchdog):
+    if watchdog is None:
+        return None
+
+    module = watchdog.get('module')
+    device_exists = WATCHDOG_DEV.exists()
+
+    # Require a usable watchdog: either device already present or a module provided
+    if not module and not device_exists:
+        raise ConfigError(
+            "No watchdog device found at /dev/watchdog0 and no module configured. "
+            "Either configure 'system watchdog module <name>' or ensure the kernel auto-loads a watchdog driver."
+        )
+
+    # If a module is provided, verify it resolves via modprobe (dry-run)
+    if module:
+        # Dry-run modprobe (-n) in quiet mode (-q) verifies availability without loading
+        rc = run(f'modprobe -n -q {module}')
```

Comment:
Only issue here is that it doesn't verify the specified module is a valid watchdog driver.

As per @dmbaturin's comment, probably best to build a validator or list of watchdog modules enabled in the VyOS kernel.